src/lib/worker.js

```diff
@@ -107,39 +108,24 @@ var inReplicaSet = function(db, pods, status, done) {
   //If we're already in a rs and NO ONE is a primary, elect someone to do the work for a primary
   var members = status.members;
 
-  var primaryExists = false;
-  for (var i in members) {
-    var member = members[i];
-
-    if (member.state === 1) {
-      if (member.self) {
-        return primaryWork(db, pods, members, false, done);
-      }
-
-      primaryExists = true;
-      break;
+  for (var member of members) {
+    if (member.state === 1 && member.self) {
+      return primaryWork(db, pods, members, false, done);
```

Comment:
If no member satisfies the condition, who will do the primary work?

Reply:
nobody, and this condition doesn't exist.